src/PlotDock.cpp

```diff
@@ -226,13 +240,22 @@ void PlotDock::updatePlot(SqliteTableModel* model, BrowseDataTableSettings* sett
                 columnitem->setText(PlotColumnType, tr("Numeric"));
 
                 // restore previous check state
-                if(mapItemsY.contains(columnitem->text(PlotColumnField)))
+                if(mapItemsY1.contains(columnitem->text(PlotColumnField)))
                 {
-                    columnitem->setCheckState(PlotColumnY, mapItemsY[columnitem->text(PlotColumnField)].active ? Qt::Checked : Qt::Unchecked);
-                    columnitem->setBackgroundColor(PlotColumnY, mapItemsY[columnitem->text(PlotColumnField)].colour);
+                    columnitem->setCheckState(PlotColumnY1, mapItemsY1[columnitem->text(PlotColumnField)].active ? Qt::Checked : Qt::Unchecked);
+                    columnitem->setBackgroundColor(PlotColumnY1, mapItemsY1[columnitem->text(PlotColumnField)].colour);
                 } else {
-                    columnitem->setCheckState(PlotColumnY, Qt::Unchecked);
+                    columnitem->setCheckState(PlotColumnY1, Qt::Unchecked);
                 }
+
+                if(mapItemsY2.contains(columnitem->text(PlotColumnField)))
+                  {
+                    columnitem->setCheckState(PlotColumnY2, mapItemsY2[columnitem->text(PlotColumnField)].active ? Qt::Checked : Qt::Unchecked);
+                    columnitem->setBackgroundColor(PlotColumnY2, mapItemsY2[columnitem->text(PlotColumnField)].colour);
+                  } else {
+                  columnitem->setCheckState(PlotColumnY2, Qt::Unchecked);
+                }
+
```

Comment:
Haven't thought much about it, but maybe there is an easy way to avoid this code duplications by having some function calls, where the appropriate Y axis is passed. It would easy future maintenance, but I don't consider it an impediment for merging. In fact, it could risk a greater risk for introducing some bug if we want to release this with the imminent 3.12 release. I'd like to know what @MKleusberg thinks about it.

Reply:
Yep, there is a lot of code duplication. I had earlier decided against optimizing this because

- I wanted to keep the style same
- Since the secondary y axis is completely independent entity, keeping it totally separate might help in the future for selective modifications.

However, I'll push a commit with the code optimizations.

Reply:
@mgrojo I'm fine either way but tend towards avoiding the duplication even when including it in the 3.12 release because duplicated code could contain hard-to-spot copy & paste errors too.

Reply:
I like the code refactoring. It's more maintainable and easier to read now. I've also tested it and found no issue. I consider it ready for merging but will wait some time so others have time to review. I think it could be merged and released in 3.12.

Thanks @anandsanto!

Reply:
I'm a huge fan of this project and what you guys do here - particularly because I have been using DB4S extensively for a while now. I'm happy that I could make a minimal contribution to this project :)